misk-crypto/src/main/kotlin/misk/crypto/CryptoModule.kt

```diff
@@ -5,35 +5,50 @@ import com.google.crypto.tink.JsonKeysetReader
 import com.google.crypto.tink.KeysetHandle
 import com.google.crypto.tink.KmsClient
 import com.google.crypto.tink.aead.AeadConfig
+import com.google.inject.Inject
+import com.google.inject.Provider
 import com.google.inject.name.Names
 import misk.config.Secret
 import misk.inject.KAbstractModule
+import misk.inject.asSingleton
 
 /**
  * Configures and registers the keys listed in the configuration file.
  * Each key is read, decrypted, and then bound via Google Guice and added to the [KeyManager].
  */
 class CryptoModule(
-  private val config: CryptoConfig,
-  private val kmsClient: KmsClient
+  private val config: CryptoConfig
 ) : KAbstractModule() {
 
   override fun configure() {
+    requireBinding(KmsClient::class.java)
     check(!(config.gcp_key_uri != null && config.aws_kms_key_alias != null))
     check(!(config.gcp_key_uri == null && config.aws_kms_key_alias == null))
     AeadConfig.register()
-    val keyManager = KeyManager()
-    bind<KeyManager>().toInstance(keyManager)
+
+    bind<KeyManager>().asSingleton()
```

Comment:
nit: @Singleton annotation is a behavior and should be on the class.

Reply:
The `KeyManager` class is also annotated with `@Singleton`

Reply:
Then why are you adding `bind<>()` here? If it has an @Inject you don't need to do this.

Reply:
Yep, totally right. My bad. 
Fixed :)